src/main/java/de/tum/in/www1/artemis/service/BitbucketService.java

```diff
@@ -107,16 +107,24 @@ public URL getRepositoryWebUrl(Participation participation) {
 
     private String getProjectKeyFromUrl(URL repositoryUrl) {
         // https://ga42xab@repobruegge.in.tum.de/scm/EIST2016RME/RMEXERCISE-ga42xab.git
-        return repositoryUrl.getFile().split("/")[2];
+        String[] urlParts = repositoryUrl.getFile().split("/");
+        if (urlParts.length > 2) {
+            return urlParts[2];
+        }
+        return "";
     }
 
     private String getRepositorySlugFromUrl(URL repositoryUrl) {
         // https://ga42xab@repobruegge.in.tum.de/scm/EIST2016RME/RMEXERCISE-ga42xab.git
-        String repositorySlug = repositoryUrl.getFile().split("/")[3];
-        if (repositorySlug.endsWith(".git")) {
-            repositorySlug = repositorySlug.substring(0, repositorySlug.length() - 4);
+        String[] urlParts = repositoryUrl.getFile().split("/");
+        if (urlParts.length > 3) {
+            String repositorySlug = urlParts[3];
+            if (repositorySlug.endsWith(".git")) {
+                repositorySlug = repositorySlug.substring(0, repositorySlug.length() - 4);
+            }
+            return repositorySlug;
         }
-        return repositorySlug;
+        return "";
     }
```

Comment:
what happens in the other code parts, in case we return "" here?
Of course we avoid an exception, but do we have any side effects when using an empty string? (sometimes an exception might be better)

Reply:
Same as above.